rules/DowngradePhp70/Rector/FuncCall/DowngradeDefineArrayConstantRector.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\DowngradePhp70\Rector\FuncCall;
+
+use PhpParser\Node;
+use PhpParser\Node\Const_;
+use PhpParser\Node\Expr\FuncCall;
+use Rector\Core\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+use PhpParser\Node\Expr\Array_;
+use PhpParser\Node\Name;
+use PhpParser\Node\Stmt\Const_ as StmtConst_;
+use PhpParser\Node\Stmt\Expression;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+
+/**
+ * @see \Rector\Tests\DowngradePhp70\Rector\Declare_\DowngradeDefineArrayConstantRector\DowngradeDefineArrayConstantRectorTest
+ */
+final class DowngradeDefineArrayConstantRector extends AbstractRector
+{
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [FuncCall::class];
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Change array contant definition via define to const',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+define('ANIMALS', [
+    'dog',
+    'cat',
+    'bird'
+]);
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+const ANIMALS = [
+    'dog',
+    'cat',
+    'bird'
+];
+CODE_SAMPLE
+                ),
+            ]
+        );
+    }
+
+    /**
+     * @param FuncCall $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if ($this->shouldSkip($node)) {
+            return null;
+        }
+
+        $parent = $node->getAttribute(AttributeKey::PARENT_NODE);
+        if (! $parent instanceof Expression) {
+            return null;
+        }
+
+        $parent = new StmtConst_(
+            [
+                new Const_(
+                    new Name(
+                        $node->args[0]->value->value
+                    ),
+                    $node->args[1]->value
+                )
+            ]
+        );
+        return $parent->consts[0];
```

Comment:
@TomasVotruba do you have idea to create `const` assign definition? The code above define constant but without `const ` start, eg:

```diff
-const ANIMALS = [
+ANIMALS = [
     'dog',
     'cat',
     'bird'
```

Reply:
it works with string `const ` prefix on const name definition https://github.com/rectorphp/rector/pull/6270/commits/73001fa5d1657f8a43cc7c4b56871b926224f47e :tada:

Reply:
that's not add "const " prefix definiton, that's solved with add "const " on name instantiation.

Reply:
Sorry, wrong link: https://github.com/rectorphp/php-parser-nodes-docs#phpparsernodestmtclassconst

Reply:
that's class constant, not a "const NAMECONST = [...]" that is not a class constant, the define is define constant that not a class constant.

Reply:
Ah, sorry. This one then: https://github.com/nikic/PHP-Parser/blob/master/lib/PhpParser/Node/Stmt/Const_.php

Reply:
I tried that previously, see https://github.com/rectorphp/rector/pull/6270/commits/dc6934f3c04bc98107b97d8a0131adcc73d93940 , the solution is add string "const " during Const_ name instantiation

Reply:
I've just pushed it. Feel free to re-use

Reply:
ok, implemented at https://github.com/rectorphp/rector/pull/6270/commits/89dcb1ea255d6ca47990857827dcc3d05cba9e0e